test/lit/source-map-names.wast

```diff
@@ -0,0 +1,24 @@
+;; RUN: wasm-opt %s --remove-unused-module-elements -o %t.wasm -osm %t.map
+;; RUN: wasm-dis %t.wasm --source-map %t.map | filecheck %s --check-prefix OUT-WAST
+;; RUN: cat %t.map | filecheck %s --check-prefix OUT-MAP
+
+;; After --remove-unused-module-elements, the output source map's 'names' field
+;; should NOT contain 'unused'
+
+;; OUT-MAP: "names":["used"]
+
+(module
+  (export "used" (func $used))
+
+  (func $unused
+    ;;@ src.cpp:1:1:unused
+    (nop)
+  )
+
+  (func $used
+    ;; OUT-WAST:      ;;@ src.cpp:2:1:used
+    ;; OUT-WAST-NEXT: (nop)
+    ;;@ src.cpp:2:1:used
+    (nop)
+  )
+)
```

Comment:
Perhaps add another used symbol, to get some coverage of the order of the symbols being deterministic? (not great coverage, but it might help)

Reply:
Done: https://github.com/WebAssembly/binaryen/pull/8068/commits/a55a84461086db11691e309b5ffbf2dacae2d48b